pkg/apis/management.cattle.io/v3/cluster_types.go

```diff
@@ -75,6 +75,10 @@ const (
 	ClusterConditionServiceAccountSecretsMigrated        condition.Cond = "ServiceAccountSecretsMigrated"
 	ClusterConditionHarvesterCloudProviderConfigMigrated condition.Cond = "HarvesterCloudProviderConfigMigrated"
 
+	// ClusterConditionHelmReleasesMigrated is true when the migration for Helm release secrets has already happened in
+	// a cluster
+	ClusterConditionHelmReleasesMigrated condition.Cond = "HelmReleasesMigrated"
```

Comment:
In my opinion, having a job brings security implications I think would be difficult to deal with (as this job has to have access to editing secrets in all namespaces).

I was examining the code and having a controller on secrets would trigger the change only when a new secret was created or updated. If no secrets were added or changed, then it would not trigger. Or did I understand this incorrectly?

To me, it doesn't feel like this process should happen in an ongoing basis. Rather, this feels to me like a one-off run to fix problems that might have happened due to a cluster upgrade.

I didn't feel like running this on each release when it is installed would be a good approach, since when an installation happens it is expected that the resources will be compatible with the cluster version. I also think it is a bit strange to edit a release that was just installed.

Reply:
I forgot to mention this but this condition was created specifically for this change. I can rename it if `HelmReleasesMigrated` is ambiguous/unclear/not a great name.

Reply:
> I was examining the code and having a controller on secrets would trigger the change only when a new secret was created or updated. If no secrets were added or changed, then it would not trigger. Or did I understand this incorrectly?

Occasionally there will be a list call that will refresh the cache (every 10 hour iirc?) but yes!

> To me, it doesn't feel like this process should happen in an ongoing basis. Rather, this feels to me like a one-off run to fix problems that might have happened due to a cluster upgrade.

If this is a one-off, then IMO a controller that exists just to perform the upgrade is too heavyweight since we'd be triggering this handler on every cluster create / update, even if we fail out early from the handler due to the condition already being met.

Reply:
I'm not sure I agree a handler is too heavy. I think adding logic to the currently existing handler is fine. It's one part of a larger handler. If other migrations exist in the future we can easily change this code to facilitate that. I'm not sure how introducing a new api endpoint is any better. Usually the debate that is happens due to this kind of logic is between putting it in a handler and start up. Both have their drawbacks.